applications/solvers/additiveFoam/movingHeatSource/refinementControllers/refinementController/refinementController.H

```diff
@@ -0,0 +1,192 @@
+    \\  /    A nd           | Copyright (C) 2011-2022 OpenFOAM Foundation
+     \\/     M anipulation  |
+-------------------------------------------------------------------------------
+                Copyright (C) 2023 Oak Ridge National Laboratory                
+-------------------------------------------------------------------------------
+License
+    This file is part of OpenFOAM.
+
+    OpenFOAM is free software: you can redistribute it and/or modify it
+    under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    OpenFOAM is distributed in the hope that it will be useful, but WITHOUT
+    ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+    FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+    for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with OpenFOAM.  If not, see <http://www.gnu.org/licenses/>.
+
+Class
+    Foam::refinementControllers::refinementController
+
+Description
+    Abstract base class for refinementController models
+
+SourceFiles
+    refinementController.C
+
+\*---------------------------------------------------------------------------*/
+
+#ifndef refinementController_H
+#define refinementController_H
+
+#include "labelVector.H"
+#include "hexMatcher.H"
+#include "dictionary.H"
+#include "volFields.H"
+#include "movingBeam.H"
+#include "treeBoundBox.H"
+#include "heatSourceModel.H"
+
+// * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //
+
+namespace Foam
+{
+
+/*---------------------------------------------------------------------------*\
+                     Class refinementController Declaration
+\*---------------------------------------------------------------------------*/
+
+class refinementController
+:
+    public IOdictionary
+{
+    // Private member functions
+    
+        //- Construct the base IO object
+        IOobject createIOobject
+        (
+            const dictionary& dict,
+            const fvMesh& mesh
+        ) const;
+    
+protected:
+
+    // Protected data
+    
+        //- PtrList to heatSourceModels
+        const PtrList<heatSourceModel>& sources_;
+        
+        //- Main heat source dictionary
+        const dictionary& heatSourceDict_;
+        
+        //- Pointer to mesh information
+        const fvMesh& mesh_;
+        
+        //- Sub-dictionary for refinementController entries
+        dictionary refinementDict_;
+        
+        //- Switch for refinementField generation
+        const bool refine_;
+        
+        //- Marker field for refinement operations
+        volScalarField refinementField_;
+        
+        //- Switch to fully resolve melt pool
+        const bool resolveTail_;
+        
+        //- Time after solidification to maintain resolved mesh
+        const scalar persistence_;
```

Comment:
The original intent of the persistence flag to prevent issues with the thermodynamic coupling when a cell coarsens. I think it is necessary to resolve the tail, but we should check if these numerical issues are solved by forcing an implicit solve whenever the mesh changes.

Reply:
Noted this in issue #14